python/src/trezorlib/ui.py

```diff
@@ -172,6 +172,52 @@ def get_passphrase(self, available_on_device: bool) -> Union[str, object]:
                 raise Cancelled from None
 
 
+class ScriptUI:
+    """Interface to be used by scripts, not directly by user.
+
+    Communicates with a client application using print() and input().
+
+    Lot of `ClickUI` logic is outsourced to the client application, which
+    is responsible for supplying the PIN and passphrase.
+
+    Reference client implementation can be found under `cli/test_script_client.py`.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def button_request(self, br: messages.ButtonRequest) -> None:
+        name = br.code.name if br.code else None
+        print(f"?BUTTON code={br.code} name={name} pages={br.pages}")
+
+    def get_pin(self, code: Optional[PinMatrixRequestType] = None) -> str:
+        print(f"?PIN code={code}")
```

Comment:
this should be `?PIN` if code is None and `?PIN code={code.name}` otherwise

Reply:
If this `UI` is to be used by scripts, should we send `code.name` (a human readable string) instead of just `code` (an int)? The second one allows for easy parsing on the client (`if int(code) == PinMatrixRequestType.Current`). 

But yeah, even in the first case, there can be probably a comparison `if code == PinMatrixRequestType.Current.name` and client can directly print it to the user

Reply:
It's an enum so we should be exposing the _symbol_, not the wire value. If you want to get back the enum value (in Python), you can do `PinMatrixRequestType.__members__[code]`. But the scripts might not be Python and might not have the enum library available -- in which case it is much nicer to rely on "human-readable" symbols than on magic numbers.

Also what I expect the script author to do is `if code == "Current"` without messing with the enum definition

Reply:
Yes, that is indeed  better. Changed in e5670a4